crates/oxc_linter/src/rules/deepscan/bad_comparison_sequence.rs

```diff
@@ -0,0 +1,197 @@
+use oxc_ast::{
+    ast::{BinaryOperator, Expression},
+    AstKind, Span,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Bad comparison sequence")]
+#[diagnostic(
+    severity(warning),
+    help(
+        "Comparison result should not be used directly as an operand of another comparison. If you need to compare three or more operands, you should connect each comparison operation with logical AND operator (`&&`)"
+    )
+)]
+struct BadComparisonSequenceDiagnostic(#[label] pub Span);
+
+/// `https://deepscan.io/docs/rules/bad-comparison-sequence`
+#[derive(Debug, Default, Clone)]
+pub struct BadComparisonSequence;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule applies when the comparison operator is applied two or more times in a row.
+    ///
+    /// ### Why is this bad?
+    /// Because comparison operator is a binary operator, it is impossible to compare three or more operands at once.
+    /// If comparison operator is used to compare three or more operands, only the first two operands are compared and the rest is compared with its result of boolean type.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// if (a == b == c) {
+    ///  console.log("a, b, and c are the same");
+    /// }
+    /// ```
+    BadComparisonSequence,
+    correctness
+);
+
+impl Rule for BadComparisonSequence {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::BinaryExpression(expr) = node.get().kind() 
+            && is_bad_comparison(node) 
+            && has_no_bad_comparison_in_parent(node, ctx) 
+        {
+            ctx.diagnostic(BadComparisonSequenceDiagnostic(expr.span));
+        }
+    }
+}
+
+fn has_no_bad_comparison_in_parent<'a, 'b>(
```

Comment:
I want to avoid catching a cascaded error twice. But not sure whether it is a good way.